tests/subsys/canbus/isotp/conformance/src/main.c

```diff
@@ -904,7 +948,12 @@ ZTEST(isotp_conformance, test_sender_fc_errors)
 	fc_frame.data[0] = FC_PCI_BYTE_1(3);
 	fc_frame.data[1] = FC_PCI_BYTE_2(fc_opts.bs);
 	fc_frame.data[2] = FC_PCI_BYTE_3(fc_opts.stmin);
+#ifdef CONFIG_ISOTP_REQUIRE_RX_PADDING
+	memset(&fc_frame.data[3], 0xCC, CAN_DL - 3);
+	fc_frame.length = CAN_DL;
+#else
 	fc_frame.length = DATA_SIZE_FC;
+#endif
```

Comment:
This particular piece of code is used no less than 8 times in this patch. Please split it out into either a macro or a function.

Reply:
@henrikbrixandersen fair enough, I'll update this

My two cents:
Keep in mind that this is unit test code. In tests like these if you remove enough duplication you often end up just rewriting the implementation that you are testing. I think the original author found a decent middle ground. Although, if I were writing these tests from scratch, all of the expected tx/rx frames would be hardcoded